pwndbg/dbg/lldb.py

```diff
@@ -9,22 +13,363 @@
+    def cast(self, type: pwndbg.dbg_mod.Type | Any) -> pwndbg.dbg_mod.Value:
+        assert isinstance(type, LLDBType)
+        t: LLDBType = type
+
+        return LLDBValue(self.inner.cast(t.inner))
+
+        return LLDBValue(self.inner.Cast(type.inner))
+
+
+class LLDBProcess(pwndbg.dbg_mod.Process):
+    def __init__(self, process: lldb.SBProcess, target: lldb.SBTarget):
+        self.process = process
+        self.target = target
+
+    @override
+    def evaluate_expression(self, expression: str) -> pwndbg.dbg_mod.Value:
+        value = self.target.EvaluateExpression(expression)
+        opt_out = _is_optimized_out(value)
+
+        if not value.error.Success() and not opt_out:
+            raise pwndbg.dbg_mod.Error(value.error.description)
+
+        return LLDBValue(value)
+
+
+class LLDBCommand(pwndbg.dbg_mod.CommandHandle):
+    def __init__(self, handler_name: str, command_name: str):
+        self.handler_name = handler_name
+        self.command_name = command_name
+
+
 class LLDB(pwndbg.dbg_mod.Debugger):
+    exec_states: List[lldb.SBExecutionState]
+
     @override
     def setup(self, *args):
+        self.exec_states = []
+
         debugger = args[0]
         assert (
             debugger.__class__ is lldb.SBDebugger
         ), "lldbinit.py should call setup() with an lldb.SBDebugger object"
 
+        module = args[1]
+        assert module.__class__ is str, "lldbinit.py should call setup() with __name__"
+
+        self.module = module
         self.debugger = debugger
 
+        import pwndbg.commands
+
+        pwndbg.commands.load_commands()
+
+        import argparse
+
+        parser = argparse.ArgumentParser(description="Prints a test message.")
+
+        @pwndbg.commands.ArgparsedCommand(parser)
+        def test2():
+            print("Test 2!")
+
+    @override
+    def add_command(
+        self, command_name: str, handler: Callable[[pwndbg.dbg_mod.Debugger, str, bool], None]
+    ) -> pwndbg.dbg_mod.CommandHandle:
+        debugger = self
+
+        class CommandHandler:
+            def __init__(self, debugger, _):
+                pass
+
+            def __call__(self, _, command, exe_context, result):
+                debugger.exec_states.append(exe_context)
+                handler(debugger, command, True)
+                assert (
+                    debugger.exec_states.pop() == exe_context
+                ), "Execution state mismatch on command handler"
+
+        # LLDB is very particular with the object paths it will accept. It is at
+        # its happiest when its pulling objects straight off the module that was
+        # first imported with `command script import`, so, we install the class
+        # we've just created as a global value in its dictionary.
+        rand = round(random.random() * 0xFFFFFFFF) // 1
+        rand = f"{rand:08x}"
+
+        name = f"__{rand}_LLDB_COMMAND_{command_name}"
+        print(f"adding command {command_name}, under the path {self.module}.{name}")
```

Comment:
I don't really get the name randomization here. Aren't all module+command combinations we create unique? Why do we need the random part at all? What is a counter-example for this?

Reply:
(This is not clear from the comment too)

Reply:
> Aren't all module+command combinations we create unique?

I'm not really sure, so, seeing as this is not hot code, I went with the safer option of making sure that, no matter what we did, these names would be very unlikely to collide. If you're sure that they're supposed to be unique, I can just take the randomization out.

Reply:
Don't we have a system in place to prevent duplicates? e.g. if some1 registers command A twice, it should fail  anyway.

I think the random can be removed